app/controllers/manifest_uploads_controller.rb

```diff
@@ -1,5 +1,46 @@
 class ManifestUploadsController < ApplicationController
   def new
+  end
+
+  def create
+    manifest = find_or_initialize_manifest
+
+    if manifest.save
+      flash[:notice] = "Upload for manifest #{manifest.tracking_number} submitted successfully."
+      redirect_to root_path
+    end
+  end
+
+  private
+
+  def find_or_initialize_manifest
+    if params[:manifest_id]
```

Comment:
We are calling this param name `id` in the API context. Does it matter if we are inconsistent?